NAMESPACE

```diff
@@ -104,6 +104,8 @@ if (getRversion() >= "4.0.0") {
   # version of R (and that is checked in .onLoad with error if not).
   export(.rbind.data.table) # only export in R<4.0.0 where it is still used; R-devel now detects it is missing doc, #5600
 }
+if (getRversion() >= "3.6.0") S3method(knitr::knit_print, data.table)
```

Comment:
what happens if knitr is not installed?

IIUC this is an Enhances dependency; weve done something similar for {lintr}

https://github.com/r-lib/lintr/blob/1db7cc70655a41c77fcc89dbed5e40d5dd8cf74a/R/zzz.R#L337

Reply:
`data.table` still loads with no warnings. `R CMD check` fails even with `_R_CHECK_FORCE_SUGGESTS_=FALSE` because it's the `VignetteBuilder`.

[WRE](https://cran.r-project.org/doc/manuals/R-exts.html#Registering-S3-methods) says:

> function `gen.cls` will get registered as an S3 method for class `cls` and generic `gen` from package `pkg` only when the namespace of pkg is loaded

Here's the code:

https://github.com/r-devel/r-svn/blob/320c6deb8bb07f9f83b3aa6cb3ca9b572c5d0b00/src/library/base/R/namespace.R#L1730-L1753

Reply:
> even with _R_CHECK_FORCE_SUGGESTS_=FALSE because it's the VignetteBuilder.

Right, we need the flags to skip vignettes too, e.g. `--no-build-vignettes` `--no-vignettes`

https://github.com/r-devel/r-svn/blob/e1cbaee23a2c12c8faa85f91aa1cc19fe709ca03/src/library/tools/R/check.R#L6851-L6852

Anyway, my concern was about loading {data.table}, thanks for the WRE cite!

I still think we should put {knitr} into `Enhances:`, WDYT?

Reply:
On the one hand, WRE says

> the `Enhances` field lists packages “enhanced” <...> by providing methods for classes from these packages

...which is a good argument for `Enhances`, but then it says

> A package should be listed in only one of these fields. 

> The `VignetteBuilder` <...> may include the current package, or ones listed in `Depends`, `Suggests` or `Imports`.

If we switch to `litedown` as the vignette builder, `knitr` should definitely go into `Enhances`. Currently, it looks like `knitr` and `markdown` should be in both `Suggests` and `VignetteBuilder`, although the latter doesn't seem to be tested. (E.g. [`cmocean` had `VignetteBuilder: knitr` for 5 years](https://github.com/aitap/cmocean/commit/7a8a907954b590455c41c5c38a1c4a98f4800fca) despite having [`%\VignetteEngine{knitr::rmarkdown}`](https://github.com/aitap/cmocean/blame/47b75447ed7ba4552952e5fc43e37261c72eb58a/vignettes/cmocean.Rmd#L7) and not a single test flagged it.)

Although it's of course possible to read WRE differently, in which case `knitr` could move from `Suggests` to `Enhances`.

Reply:
You're 100% right, I'm not sure what I was thinking. {knitr} is already in `Suggests`, which is a stronger dependency than `Enhances`. #6583 is to use {litedown} instead, I added a comment there about being sure to register `Enhances` if {knitr} is dropped from `Suggests`.